File: cinema-backend/src/main/java/org/example/cinemabackend/movie/application/dto/request/ProductionDetailsRequest.java
package org.example.cinemabackend.movie.application.dto.request;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.Builder;

import java.time.LocalDate;
import java.util.Set;

@Builder
public record ProductionDetailsRequest(
        @NotNull LocalDate worldPremiereDate,
        @NotNull @Valid FilmMemberRequest director,
        @NotNull Set<@NotNull @Valid FilmMemberRequest> actors,
        @NotNull Set<String> originalLanguages,
        @NotNull Set<String> productionCountries
) {
}

File: cinema-frontend/src/app/movie/dtos/request/production-details.request.ts
import {FilmMemberRequest} from "./film-member.request";

export class ProductionDetailsRequest {
  constructor(
    readonly worldPremiereDate: Date,
    readonly director: FilmMemberRequest,
    readonly actors: FilmMemberRequest[],
    readonly originalLanguages: string[],
    readonly productionCountries: string[],
  ) {
  }
}
